tox.ini

```
@@ -0,0 +1,73 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = install, lint, snap, unit
+
+[vars]
+snap_path = {toxinidir}/snap/
+tst_path = {toxinidir}/tests/
+all_path = {[vars]tst_path}
+
+[testenv]
+setenv =
+    PYTHONPATH = {toxinidir}:{[vars]tst_path}
+    PYTHONBREAKPOINT=ipdb.set_trace
+    PY_COLORS=1
+passenv =
+    PYTHONPATH
+allowlist_externals =
+    /snap/bin/snapcraft
+    /usr/bin/snap
+    bash
+    rm
+    sudo
```

Usually I specify the full path so I know what specific binary I am allowing into the test environment:

```suggestion
    /usr/bin/bash
    /usr/bin/rm
    /usr/bin/sudo
```